src/lib/modules/job/job.go

```diff
@@ -0,0 +1,189 @@
+package job
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	config "github.com/bench-routes/bench-routes/src/lib/config_v2"
+	"github.com/bench-routes/bench-routes/src/lib/modules/evaluate"
+	"github.com/bench-routes/bench-routes/tsdb/file"
+)
+
+// Executable is an interface that is implemented by machineJob and monitoringJob
+type Executable interface {
+	Execute()
+	Abort()
+	Info() *JobInfo
+}
+
+type JobInfo struct {
+	mux         sync.RWMutex
+	Name        string
+	Every       time.Duration
+	lastExecute time.Time
+}
+
+// NewJob creates a new job based on the typ
+func NewJob(typ string, app file.Appendable, c chan struct{}, api *config.API) (Executable, error) {
```

Comment:
```suggestion
func NewJob(typ string, app file.Appendable, c <-chan struct{}, api *config.API) (Executable, error) {
```
c is always an output channel.

Reply:
If we are passing only the read-only channel in the NewJob function which will be eventually used by monitoring and machine jobs, we would not be able to close the channel in Abort() function as it is read-only.

Reply:
```suggestion
func NewJob(typ string, app file.Appendable, api *config.API) (Executable, chan<- struct{}, error) {
```